tools/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/handler/AbstractQueryRequestHandler.java

```diff
@@ -114,17 +112,25 @@ public ModelAndView handleQueryRequest(HttpServletRequest request, RequestMethod
 			}
 
 		} catch (Exception e) {
-			// only close the connection when an exception occurs. Otherwise, the QueryResultView will take care of
-			// closing it.
-			if (repositoryCon != null) {
-				repositoryCon.close();
+			// only close the response & connection when an exception occurs. Otherwise, the QueryResultView will take
+			// care of closing it.
+			try {
+				if (queryResponse != null) {
+					queryResponse.close();
+				}
+			} catch (Exception qre) {
+				logger.warn("Query response closing error", qre);
```

Comment:
Question: I thought it might be sensible here to log the result-close failure so the client still receives the original failure reason. Is that the right approach? And: Would it make sense to do the same for `repositoryCon` below?

Reply:
Ifb01e2a2681ddbd150f